packages/@lwc/template-compiler/src/shared/renderer-hooks.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { invariant, TemplateErrors } from '@lwc/errors';
+import State from '../state';
+import { BaseElement, ElementDirective } from './types';
+
+/**
+ * Config representing criteria for an element match.
+ * All conditions specified must be satisfied to be considered a match.
+ */
+export interface CustomRendererElementConfig {
+    /**
+     * Tag name to use to match an element.
+     */
+    tagName: string;
```

Comment:
Would it make sense to make this optional, in case there is a situation where we want to match some attribute/attributes and we don't care what the tag name is?

Reply:
In the future, that is a possibility. `href` would be a good example of such an attribute. 
Having said that, the requirement can be relaxed. That change will also be back wards compatible. IMO, we should go with a very explicit requirement and loosed it if required. Additionally:
1. The initial list of elements that require the hooks is quite small and they are tag name specific.
2. Also matching attributes of each element parsed would be less performant than matching tagName as a minimum requirement.